sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/HiveFileFormat.scala

```diff
@@ -133,7 +133,7 @@ class HiveOutputWriter(
   private val standardOI = ObjectInspectorUtils
     .getStandardObjectInspector(
       tableDesc.getDeserializer(jobConf).getObjectInspector,
-      ObjectInspectorCopyOption.JAVA)
+      ObjectInspectorCopyOption.DEFAULT)
```

Comment:
Can you point to me where we convert `UTF8String` to string or bytes according to this mode? `UTF8String` is a Spark class and I assume this part of code should be in Spark?

Reply:
The code is at `HiveInspectors.wrapperFor()`.
```
case _: StringObjectInspector if x.preferWritable() =>
        withNullSafe(o => getStringWritable(o))
case _: StringObjectInspector =>
        withNullSafe(o => o.asInstanceOf[UTF8String].toString())
```

Before write data with hiveWriter, spark will use `HiveInspectors.wrapperFor()` to convert `UTF8String` to `Text` or `String`.

Reply:
ah I see. Can you add some comments here? Saying that the `DEFAULT` mode indicates Spark to create Text with the bytes of `UTF8String`.

Reply:
OK. I will add it later.